esrally/telemetry.py

```diff
@@ -92,9 +92,12 @@ def detach_from_node(self, node, running):
     def on_benchmark_start(self):
         for device in self.devices:
             if self._enabled(device):
+                if self.serverless_mode and not self._available_on_serverless(device):
+                    if device.command in self.enabled_devices:
```

Comment:
I find this logic confusing. Isn't this check redundant given that `self._enabled` checks this already?